### PR TITLE
refactor(AccordionItem): ｍove `AccordionItem` logic to `use-accordion-item.ts`

### LINKED ISSUE #3347: Separation of UI and Logic in Component `AccordionItem`
### Description

Improving Code Maintainability and Readability.

### Problem Statement/Justification

Separate the logic in `AccordionItem` component into a use-accordion-item.ts file.

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

Yes

### Additional Information

Create a new branch from 2.0 branch for development.

### PR DESCRIPTION
**Pull request by kaehehehe**
Closes #3347

## Description

Separate the logic in AccordionItem component into a use-accordion-item.ts file.

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Review by odendayoko — COMMENTED** [review]
@kaehehehe @hirotomoyamada 
Thanks!! 
May I confirm one thing?

"Logic like states, props, etc. should be ported to hooks" related Issue, code related to UI should not be cut out for hooks, right?
In this case, it is correct to say that lines 173~193 of `accordion-item.tsx` do not need to be changed, right?

```ts :accordion-item.tsx
const css: CSSUIObject = { ...styles.item, overflowAnchor: "none" }

const cloneLabel =
  typeof label === "function"
    ? label({
        isDisabled,
        isExpanded: isOpen,
      })
    : label

if (typeof children === "function")
  children = children({ isDisabled, isExpanded: isOpen })

const validChildren = getValidChildren(children)

const customAccordionLabel = findChild(validChildren, AccordionLabel)
const customAccordionPanel = findChild(validChildren, AccordionPanel)

const cloneChildren = !isEmpty(validChildren)
  ? omitChildren(validChildren, AccordionLabel, AccordionPanel)
  : children
```

**Comment by kaehehehe** [comment]
> @kaehehehe @hirotomoyamada Thanks!! May I confirm one thing?
> 
> "Logic like states, props, etc. should be ported to hooks" related Issue, code related to UI should not be cut out for hooks, right? In this case, it is correct to say that lines 173~193 of `accordion-item.tsx` do not need to be changed, right?
> 
> ```ts
> const css: CSSUIObject = { ...styles.item, overflowAnchor: "none" }
> 
> const cloneLabel =
>   typeof label === "function"
>     ? label({
>         isDisabled,
>         isExpanded: isOpen,
>       })
>     : label
> 
> if (typeof children === "function")
>   children = children({ isDisabled, isExpanded: isOpen })
> 
> const validChildren = getValidChildren(children)
> 
> const customAccordionLabel = findChild(validChildren, AccordionLabel)
> const customAccordionPanel = findChild(validChildren, AccordionPanel)
> 
> const cloneChildren = !isEmpty(validChildren)
>   ? omitChildren(validChildren, AccordionLabel, AccordionPanel)
>   : children
> ```



> 173~193 of `accordion-item.tsx`

@odendayoko 
I intentionally chose not to move the CSS-related logic into the hook. I understand that the purpose of this refactoring is to separate logic from style and improve the readability of the code. Therefore, I believe that including the code from lines 173 to 193 of `accordion-item.tsx` in the hook might dilute the intended purpose.

**Review by odendayoko — APPROVED** [review]
@kaehehehe 
> I intentionally chose not to move the CSS-related logic into the hook. I understand that the purpose of this refactoring is to separate logic from style and improve the readability of the code. Therefore, I believe that including the code from lines 173 to 193 of accordion-item.tsx in the hook might dilute the intended purpose.

I agree with your opinion. Just to be sure, I have confirmed.
Thank you! LGTM🧡

File: packages/components/accordion/src/use-accordion-item.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import type { KeyboardEvent, KeyboardEventHandler } from "react"
+import type { AccordionItemOptions } from "./accordion-item"
+import { ariaAttr, handlerAll, isArray, mergeRefs } from "@yamada-ui/utils"
+import { useCallback, useId } from "react"
+import {
+  useAccordionContext,
+  useAccordionDescendant,
+} from "./accordion-context"
+
+interface useAccorDionItemProps
```
Comment (odendayoko) [inline packages/components/accordion/src/use-accordion-item.ts:11]: @kaehehehe 
I would like the type definitions to be written in PascalCase.Can you change it to `UseAccorDionItemProps`?

File: packages/components/accordion/src/use-accordion-item.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  extends Pick<AccordionItemOptions, "isDisabled"> {}
+
+export function useAccorDionItem({ isDisabled }: useAccorDionItemProps) {
```
Comment (odendayoko) [inline packages/components/accordion/src/use-accordion-item.ts:14]: @kaehehehe 
Here is a typo.Please correct it to `useAccordionItem`.

**Review by odendayoko — CHANGES_REQUESTED** [review]
@kaehehehe 
Sorry for requesting the change even though I approved it once🙇‍♀️
I noticed the typo, could you please correct it?

File: packages/components/accordion/src/use-accordion-item.ts (hunk 2 above)
Comment (kaehehehe) [reply packages/components/accordion/src/use-accordion-item.ts]: Thank you for catching that!

File: packages/components/accordion/src/use-accordion-item.ts (hunk 1 above)
Comment (kaehehehe) [reply packages/components/accordion/src/use-accordion-item.ts]: Fixed!

**Review by odendayoko — CHANGES_REQUESTED** [review]
@kaehehehe 
One typo was not corrected, please check🙇

**Comment by kaehehehe** [comment]
Oops! I'm sorry. I corrected the typo! @odendayoko

**Comment by hirotomoyamada** [comment]
@kaehehehe 

If this PR is incorporated first, conflicts will occur in other work, so close this PR.

Please submit this PR again after the other work is completed.

**Pull request closed without merging**